integration-tests/bats/diff.bats

```diff
@@ -869,3 +869,55 @@ SQL
    [ $status -eq 0 ]
    [[ $output =~ '| + | 1  | ramen        |' ]] || false
 }
+
+@test "diff: allowed across primary key renames" {
+    dolt sql <<SQL
+CREATE TABLE t1 (pk int PRIMARY KEY, col1 int);
+INSERT INTO t1 VALUES (1, 1);
+CREATE TABLE t2 (pk1a int, pk1b int, col1 int, PRIMARY KEY (pk1a, pk1b));
+INSERT INTO t2 VALUES (1, 1, 1);
+SQL
+    dolt commit -am "initial"
+
+    dolt sql <<SQL
+ALTER TABLE t1 RENAME COLUMN pk to pk2;
+UPDATE t1 set col1 = 100;
+ALTER TABLE t2 RENAME COLUMN pk1a to pk2a;
+ALTER TABLE t2 RENAME COLUMN pk1b to pk2b;
+UPDATE t2 set col1 = 100;
+SQL
+    dolt commit -am 'rename primary key'
+
+    run dolt diff HEAD~1 HEAD
+    [ $status -eq 0 ]
+
+    [[ "$output" =~ ' CREATE TABLE `t1` (' ]] || false
+    [[ "$output" =~ '-  `pk` int NOT NULL,' ]] || false
+    [[ "$output" =~ '+  `pk2` int NOT NULL,' ]] || false
+    [[ "$output" =~ '   `col1` int,' ]] || false
+    [[ "$output" =~ '-  PRIMARY KEY (`pk`)' ]] || false
+    [[ "$output" =~ '+  PRIMARY KEY (`pk2`)' ]] || false
+    [[ "$output" =~ ' ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_bin;' ]] || false
+    [[ "$output" =~ '+---+------+------+------+' ]] || false
```

Comment:
Maybe a little overspecified for the output checks but also fine to be lazy if you make it easy to change.

If you're gonna match large multiline text like this, use this method:

https://github.com/dolthub/dolt/blob/main/integration-tests/compatibility/test_files/bats/compatibility.bats#L135